plugin/acl/acl_test.go

```diff
@@ -205,6 +206,79 @@ func TestACLServeDNS(t *testing.T) {
 			wantRcode:             dns.RcodeRefused,
 			wantExtendedErrorCode: dns.ExtendedErrorCodeBlocked,
 		},
+		{
+			name: "Drop 1 DROPPED",
+			config: `acl example.org {
+				drop net 192.168.0.0/16
+			}`,
+			zones: []string{},
+			args: args{
+				domain:   "www.example.org.",
+				sourceIP: "192.168.0.2",
+				qtype:    dns.TypeA,
+			},
+			wantRcode:        dns.RcodeSuccess,
+			expectNoResponse: true,
+		},
+		{
+			name: "Subnet-Order 1 REFUSED",
+			config: `acl example.org {
+				block net 192.168.1.0/24
+				drop net 192.168.0.0/16
+			}`,
+			zones: []string{},
+			args: args{
+				domain:   "www.example.org.",
+				sourceIP: "192.168.1.2",
+				qtype:    dns.TypeA,
+			},
+			wantRcode:             dns.RcodeRefused,
+			wantExtendedErrorCode: dns.ExtendedErrorCodeBlocked,
+		},
+		{
+			name: "Subnet-Order 2 DROPPED",
+			config: `acl example.org {
+				drop net 192.168.0.0/16
+				block net 192.168.2.0/24
```

Comment:
I feel `block net 192.168.1.0/24` would be better here considering you used `192.168.1.1` as the source IP.

Reply:
Thanks and I agree, updated to `192.168.1.0/24`.